sdk/messaging/azservicebus/internal/tracing/tracing.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package tracing
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+)
+
+type Attribute = tracing.Attribute
+type SpanKind = tracing.SpanKind
+type Tracer = tracing.Tracer
+type Provider = tracing.Provider
+
+func NewNoOpTracer() Tracer {
+	return Tracer{}
+}
+
+type SpanConfig struct {
+	name       SpanName
+	attributes []Attribute
+}
+
+type SetAttributesFn func([]Attribute) []Attribute
+
+func NewSpanConfig(name SpanName, options ...SetAttributesFn) *SpanConfig {
```

Comment:
I think this indirection (for SetAttributesFn) is a bit odd - it looks like `attributes ...Attribute` would cover what we use it for, and helps a bit in showing that's the only thing we use it for.

Reply:
switched this one up a bit to directly return list of attributes, let me know what you think!